test/integration/swap.behavior.t.sol

```diff
@@ -0,0 +1,509 @@
+
+    function setUp() public {
+        vm.rollFork(BLOCK_NUMBER);
+
+        PERMIT2 = IPermit2(UNISWAP_PERMIT2);
+
+        Setup setup = new Setup();
+
+        (factory,, settings,) = setup.deploySystem({
+            _deployer: address(0),
+            _owner: address(this),
+            _addressResolver: ADDRESS_RESOLVER,
+            _gelato: address(0),
+            _ops: address(0),
+            _universalRouter: UNISWAP_UNIVERSAL_ROUTER,
+            _permit2: UNISWAP_PERMIT2
+        });
+
+        account = Account(payable(factory.newAccount()));
+
+        sUSD = IERC20(IAddressResolver(ADDRESS_RESOLVER).getAddress(PROXY_SUSD));
+        mintSUSD(address(this), AMOUNT);
+
+        vm.prank(EOA_WITH_DAI);
+        dai.transfer(address(this), AMOUNT);
+
+        // call approve() on an ERC20 to grant an infinite allowance to the canonical Permit2 contract
+        sUSD.approve(UNISWAP_PERMIT2, type(uint256).max);
+
+        // call approve() on the canonical Permit2 contract to grant an infinite allowance to the SM Account
+        /// @dev this can be done via PERMIT2_PERMIT in production
+        PERMIT2.approve(
+            address(sUSD), address(account), type(uint160).max, type(uint48).max
+        );
+
+        // call approve() on an ERC20 to grant an infinite allowance to the canonical Permit2 contract
+        dai.approve(UNISWAP_PERMIT2, type(uint256).max);
+
+        // call approve() on the canonical Permit2 contract to grant an infinite allowance to the SM Account
+        /// @dev this can be done via PERMIT2_PERMIT in production
+        PERMIT2.approve(
+            address(dai), address(account), type(uint160).max, type(uint48).max
+        );
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                 TESTS
+    //////////////////////////////////////////////////////////////*/
+
+    /*//////////////////////////////////////////////////////////////
+                            SINGLE POOL SWAP
+    //////////////////////////////////////////////////////////////*/
+
+    function test_UniswapV3Swap_DAI_SUSD() public {
+        // whitelist DAI
+        settings.setTokenWhitelistStatus(address(dai), true);
+
+        // define command(s)
+        IAccount.Command[] memory commands = new IAccount.Command[](1);
+        commands[0] = IAccount.Command.UNISWAP_V3_SWAP;
+
+        // define input(s)
+        bytes[] memory inputs = new bytes[](1);
+        uint256 amountIn = AMOUNT / 2;
+        uint256 amountOutMin = 1;
+        bytes memory path = bytes.concat(
+            bytes20(address(dai)), LOW_FEE_TIER, bytes20(address(sUSD))
+        );
+        inputs[0] = abi.encode(amountIn, amountOutMin, path);
+
+        uint256 preBalance = sUSD.balanceOf(address(account));
+        account.execute(commands, inputs);
+        uint256 postBalance = sUSD.balanceOf(address(account));
+
+        assertGt(postBalance, preBalance);
```

Comment:
**Nitpick:** Not really needed but would be nice to assert that the `dai` balance has been updated as expected too:
```solidity
        uint256 preBalance = sUSD.balanceOf(address(account));
        uint256 preBalanceDai = dai.balanceOf(address(this));
        account.execute(commands, inputs);
        uint256 postBalance = sUSD.balanceOf(address(account));
        uint256 postBalanceDai = dai.balanceOf(address(this));

        assertGt(postBalance, preBalance);
        assertEq(postBalanceDai, preBalanceDai / 2);
```

Reply:
not a bad idea. added these checks to rest of tests for swapping